main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -2949,20 +2950,20 @@ object Resolver {
         // Make sure we don't duplicate results in `rootNames`
         if (ns0.idents.size > 1 && ns0.idents.lastOption.contains(qname.ident)) {
           // Case 1.1.1.1: We are referring to the current namespace. Use that.
-          root.symbols.getOrElse(Name.mkUnlocatedNName(ns0.parts.init), Map.empty).getOrElse(ns0.parts.last, Nil).map(Resolution.Declaration)
+          root.symbols.getOrElse(Name.mkUnlocatedNName(ns0.parts.init), Map.empty).getOrElse(ns0.parts.last, Nil).map(Resolution.Declaration.apply)
         } else {
           Nil
         }
       }
 
       // 4th priority: names in the root namespace
-      val rootNames = root.symbols.getOrElse(Name.RootNS, Map.empty).getOrElse(qname.ident.name, Nil).map(Resolution.Declaration)
+      val rootNames = root.symbols.getOrElse(Name.RootNS, Map.empty).getOrElse(qname.ident.name, Nil).map(Resolution.Declaration.apply)
 
-      envNames ::: localNames ::: currentNamespace ::: rootNames
+      (envNames ::: localNames ::: currentNamespace ::: rootNames).distinct
```

Comment:
This change may be significant?

Reply:
Same question: Why distinct?

Reply:
I inspected the results and the code above finds duplicate resolutions, e.g., it will find `Option.None` multiple times. However, the caller(s) of this function assumes that only one lookup should be returned if successful. This is (obviously) either intentional or a bug in this function but it seems like once we have found a resolution we shouldn't return it multiple times. Maybe it should be a set instead?